src/Sarif.Viewer.VisualStudio.ResultSources.DeveloperCanvas.Core/Services/DevCanvasResultSourceService.cs

```diff
@@ -0,0 +1,384 @@
+
+                // Get the generators that we should query for this cache item.
+                List<DevCanvasGeneratorInfo> allGenerators = await accessor.GetGeneratorsAsync();
+
+                // Now query the server for insights.
+                Trace.WriteLine($"Querying for insights for {absoluteFilePath}");
+
+
+                List<SarifLog> logs = new List<SarifLog>();
+                foreach (DevCanvasGeneratorInfo generator in allGenerators)
+                {
+                    DevCanvasRequestV1 requestObject = new DevCanvasRequestV1(generator.Name, repoRootedFilePath, vcDetails);
+
+                    SarifLog log = await accessor.GetSarifLogV1Async(requestObject);
+                    // filter out the xaml presentation since we cant render it here
+                    log = TrimLog(log);
+                    logs.Add(log);
+                }
+
+                SarifLog masterLog = CombineLogs(logs);
+                foreach (Run run in masterLog.Runs)
+                {
+                    insightsCount += run.Results.Count;
+                }
+
+                // Add the number of cached insights to the result message that we'll display in the output window (after releasing the lock).
+                resultMessage.Append($"Cached {Util.S("insight", insightsCount)} for {absoluteFilePath}");
+
+                // Now that this query is complete, see if the number of outstanding queries is less than the max,
+                // and if so, try to run the next query in the queue (if any).
+                int queuedCount = 0;
+                outstandingQueries--;
+                if (outstandingQueries < maxParallelQueries)
+                {
+                    string nextFilePath = null;
+                    if (filePathQueue.Count != 0)
+                    {
+                        nextFilePath = filePathQueue.Dequeue();
+
+                        outstandingQueries++;
+                        ThreadPool.QueueUserWorkItem(QueryInsights, nextFilePath);
+                    }
+
+                    // Periodically log a message to indicate how many files are still queued for processing.
+                    queuedCount = filePathQueue.Count;
+                    if (queuedCount > 0 && (queuedCount % 100 == 0))
+                    {
+                        resultMessage.Append($"\n{queuedCount} files remaining.");
+                    }
+                }
+
+                // Display the result message (e.g. "Cached X insights for file Y")
+                Trace.WriteLine(resultMessage.ToString());
+
+                RaiseServiceEvent(new ResultsUpdatedEventArgs()
+                {
+                    SarifLog = masterLog,
+                    LogFileName = "",
+                    UseDotSarifDirectory = false
+                });
+            }
+            catch (Exception)
+            {
```

Comment:
Trace.WriteLine() the exception here?

Reply:
No, however ```async void``` methods that throw exceptions can crash VS as a whole, and so it's important to wrap them in try catches. Mostly done for future proofing since I don't think Trace.WriteLine can throw exceptions.